types/lodash/index.d.ts

```diff
@@ -3954,6 +3954,10 @@ declare namespace _ {
             arrays3: List<T> | null | undefined,
             arrays4: List<T> | null | undefined,
             arrays5: List<T> | null | undefined,
+            iteratee: (value1: T, value2: T, value3: T, value4: T, value5: T) => TResult
+        ): TResult[];
+
+        zipWith<T, TResult>(
```

Comment:
This last overload weakens all the type checking provided by the specific overloads above. For example, things like this become legal, when they would have been errors before.

```ts
zipWith(null)
zipWith((something) => {}, undefined, (randomfunction) => 3)
zipWith(list1, list2) // no iteratee parameter
zipWith(list1, list2, (a,b,c) => 3) // wrong number of parameters on iteratee
```

Unfortunately I can't think of a better solution to your problem, but I don't think this is a good change.

Reply:
Unfortunately, without a change we break a significant number of use cases within lodash. With typescript not having a proper way to provide strong typings for variadic tuples, I would rather err on the side of weak types that do not limit the usability specified within lodash's documentation. If the user does not know the number of arrays at compile time but wants to use an iteratee, they currently cannot without using `as any` everywhere.

Additionally, all of your examples are actually valid uses of `zipWith` except for the second where there are two iteratee functions. 

Common usages that aren't currently possible via these typings:
```ts
zipWith([1, 2], null, ['a', 'b'])
zipWith(...arrayOfArrays)
```

I guess a lot of this comes down to: I don't want to tell the user they can't use lodash a certain way because I want to lock down every aspect of the type safety.

Reply:
I see what you're saying. (BTW is my last example also valid? I thought there might be an issue with having the wrong number of parameters on iteratee).

I'm sympathetic to this change, but I'd like to give it some time in case the authors / other users of lodash want to review.

Reply:
Technically it would be valid, you can always provide a lambda with extra parameters, the extras will just be `undefined`. Though clearly if we can find a way to enforce that the first `n - 1` parameters are of type `Array<{} | null | undefined>` and the `n`th parameter is a function with the correct number of arguments, that would be preferable. Unfortunately, lodash wasn't really written with type safety in mind, it was written to be as permissive as possible. Our typings within DefinitelyTyped should support that decision, and leave it up to lodash (hopefully v5) to enforce some type safe practices.